omnibus/package-scripts/agent/prerm

```diff
@@ -151,7 +151,7 @@ elif [ -f "/etc/redhat-release" ] || [ -f "/etc/system-release" ] || [ -f "/etc/
         ;;
         1)
             # We're upgrading.
-            # The preinst script has taken care of removing the .pyc/.pyo files
+            remove_downloader_data
```

Comment:
Both the uninstall and upgrade paths seem to work as expected on CentOS now ✅

Reply:
You're right, I completely forgot that the scripts order is different in the rpm package, compared to the deb package.
The rationale behind this change was that we shouldn't rely on what the preinst script of a future version of a package does. This works with the deb package because you can actually run something (the prerm) before the preinst of the new package, but not with the rpm package.
But as you noted, this too has downsides, so we have to compromise between three non-ideal solutions:
- either we have to remove the downloader data for all versions of Python 3 we ever shipped in the preinst (and in the prerm too, for uninstall),
- or we use a glob in the preinst to avoid code duplication in the preinst (something like:
```
DOWNLOADER_DATA="$INSTALL_DIR/embedded/lib/python3.*/site-packages/datadog_checks/downloader/data"
for DATA_DIR in $DOWNLOADER_DATA; do
  if [ -d $DATA_DIR ]; then
    rm -rf $DATA_DIR
  fi
done
```
- or do the removal in the prerm, which implies not being able to ship anything in `/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/downloader/data`.

However, I'd say the implementation we choose in this PR is not that important, since it will be overwritten by https://github.com/DataDog/datadog-agent/pull/4811 which replaces the downloader data cleanup step. We could revert to the previous behavior (remove all downloader data in preinst + remove 3.8 downloader data in prerm) if we think not being able to ship anything in `/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/downloader/data` is too much of a problem.

Reply:
@olivielpeau 

> I don't have the full context, but I imagine that the reason you do the removal of the 3.8 datadog_checks/downloader/data directory in the prerm script instead of the preinst script on RHEL/SUSE is that, when we upgrade to python 3.9, this will allow us to simply update the 3.8 to 3.9 in the path of the directory in this prerm script and we won't have to keep the logic removing the 3.8 directory?

Correct.

After discussion with @KSerrania, because changes will be overwritten by #4811, I've reverted the behavior to the one before the PR (just added the lines to remove 38 in preinst), this way my PR only deals with the Python upgrade and not really the pkg logic.